js/modules/loki_public_chat_api.js

```diff
@@ -72,13 +79,102 @@ class LokiPublicServerAPI {
     this.channels.splice(i, 1);
     thisChannel.stopPolling = true;
   }
+
+  async getServerToken() {
+    let token = await Signal.Data.getPublicServerTokenByServerName(this.server);
+    if (!token) {
+      token = await this.getNewToken();
+      if (token) {
+        await Signal.Data.savePublicServerToken({
+          server: this.server,
+          token,
+        });
+      }
+    }
+    return token;
+  }
+
+  async getNewToken() {
+    if (!this.tokenPending) {
```

Comment:
You should be able to replace `this.tokenPending` with `this.tokenPromise !== null` and set it to `null` after it has resolved ?

Reply:
Luv it